Remove c++14-conditional constexpr

smessmer opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#30916 Remove c++14-conditional constexpr**

These macros said "make it constexpr if we're in C++14". Since we're now always C++14, we can just say "constexpr" isntead.

Differential Revision: [D18869635](https://our.internmc.facebook.com/intern/diff/D18869635/)

smessmer committed:
Remove c++14-conditional constexpr

These macros said "make it constexpr if we're in C++14". Since we're now always C++14, we can just say "constexpr" isntead.

Differential Revision: [D18869635](https://our.internmc.facebook.com/intern/diff/D18869635/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove c++14-conditional constexpr"


These macros said "make it constexpr if we're in C++14". Since we're now always C++14, we can just say "constexpr" isntead.

Differential Revision: [D18869635](https://our.internmc.facebook.com/intern/diff/D18869635/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove c++14-conditional constexpr"


These macros said "make it constexpr if we're in C++14". Since we're now always C++14, we can just say "constexpr" isntead.

Differential Revision: [D18869635](https://our.internmc.facebook.com/intern/diff/D18869635/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove c++14-conditional constexpr"


These macros said "make it constexpr if we're in C++14". Since we're now always C++14, we can just say "constexpr" isntead.

Differential Revision: [D18869635](https://our.internmc.facebook.com/intern/diff/D18869635/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove c++14-conditional constexpr"


These macros said "make it constexpr if we're in C++14". Since we're now always C++14, we can just say "constexpr" isntead.

Differential Revision: [D18869635](https://our.internmc.facebook.com/intern/diff/D18869635/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove c++14-conditional constexpr"


These macros said "make it constexpr if we're in C++14". Since we're now always C++14, we can just say "constexpr" isntead.

Differential Revision: [D18869635](https://our.internmc.facebook.com/intern/diff/D18869635/)

[ghstack-poisoned]

kostmo commented:
## CircleCI build failures summary
As of commit 8ac52b5:
* **2/2** broken upstream at merge base 643ca5d (see [grid view](https://dr.pytorch.org/master-timeline.html?count=1&sha1=643ca5def2ea140f16bb064740b8de331ee465d8&should_suppress_scheduled_builds=true&should_suppress_fully_successful_columns=true&max_columns_suppress_successful=35))
    * You may want to rebase on the `viable/strict` branch (see its [recency history](https://dr.pytorch.org/master-viable-commits.html)):
        * If your commit is newer than `viable/strict`, you can try basing on an older, stable commit:
          ```
          git fetch viable/strict
          git rebase --onto viable/strict $(git merge-base origin/master HEAD)
          ```
        * If your commit is older than `viable/strict`:
          ```
          git fetch viable/strict
          git rebase viable/strict
          ```
* **0/2** failures introduced in this PR



## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=8ac52b51d9a047116d7544f8e661652431b01150).

### 2 upstream failures recognized by patterns:
These builds matched patterns, but were probably caused by upstream breakages:
* [caffe2_onnx_py2_gcc5_ubuntu16_04_test](https://dr.pytorch.org/build-details.html?build_id=75657840)
* [caffe2_onnx_py3_6_clang7_ubuntu16_04_test](https://dr.pytorch.org/build-details.html?build_id=75661245)



---

<details><summary>This comment was automatically generated by <a href="https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment">Dr. CI</a> (expand for details).</summary>Follow <a href="https://dr.pytorch.org/admin/comments-opt-out.html">this link to opt-out</a> of these comments for your Pull Requests.

Please report bugs/suggestions on the <a href="https://github.com/kostmo/circleci-failure-tracker/issues">GitHub issue tracker</a>.
</details>

smessmer committed:
Update on "Remove c++14-conditional constexpr"


These macros said "make it constexpr if we're in C++14". Since we're now always C++14, we can just say "constexpr" isntead.

Differential Revision: [D18869635](https://our.internmc.facebook.com/intern/diff/D18869635/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove c++14-conditional constexpr"


These macros said "make it constexpr if we're in C++14". Since we're now always C++14, we can just say "constexpr" isntead.

Differential Revision: [D18869635](https://our.internmc.facebook.com/intern/diff/D18869635/)

[ghstack-poisoned]
